projects/packages/connection/src/class-manager.php

```diff
@@ -526,7 +493,16 @@ public function is_active() {
 		if ( ( new Status() )->is_no_user_testing_mode() ) {
 			return $this->is_connected();
 		}
-		return (bool) $this->get_access_token( self::CONNECTION_OWNER );
+		return (bool) $this->get_tokens()->get_access_token( true );
+	}
+
+	/**
+	 * Obtains an instance of the Tokens class.
+	 *
+	 * @return Tokens the Tokens object
+	 */
+	public function get_tokens() {
```

Comment:
Because instantiating classes is very cheap and keeping instances floating around is expensive, I guess. I don't like making things stateful unless I absolutely have to. At some point, the constructor of Tokens is going to take a user_id, and it'll have base classes and things like that (e.g. `new MasterUserTokens()` which invokes `super::tokens( true )`).

Reply:
Thanks for taking the time to elaborate on this! 👍